src/sdkBase.ts

```diff
@@ -174,33 +188,109 @@ export abstract class SdkBase {
   }
 
   private initInternal(opts: SdkOptions | unknown) {
-    if (this._initComplete) {
-      throw new TypeError('Calling init() more than once is not allowed');
-    }
     if (!isSDKOptionsOrThrow(opts))
       throw new TypeError(`Options provided to ${this._product.name} init couldn't be validated.`);
 
-    storeConfig(opts, this._product);
+    if (this._hasAborted) {
+      throw new TypeError('Calling init() once aborted or disconnected is not allowed');
+    }
 
-    this._opts = opts;
-    this._storageManager = new StorageManager(
-      { ...opts },
-      this._product.cookieName,
-      this._product.localStorageKey
-    );
-    this._apiClient = new ApiClient(opts, this._product.defaultBaseUrl, this._product.name);
-    this._tokenPromiseHandler.registerApiClient(this._apiClient);
+    if (this.isInitialized()) {
+      this.setInitComplete(false);
+
+      const previousOpts = { ...this._opts };
+      let shouldUpdateConfig = false;
+      let shouldUpdateIdentityValue = false;
+
+      if (opts.cookieDomain && opts.cookieDomain != this._opts.cookieDomain) {
+        shouldUpdateConfig = true;
+        shouldUpdateIdentityValue = true;
+        this._opts.cookieDomain = opts.cookieDomain;
+        this._logger.log('cookie domain updated');
+      }
+
+      if (opts.cookiePath && opts.cookiePath !== this._opts.cookiePath) {
```

Comment:
It's easy to misread which of these is the new options and which is the previous options. You've already got a _previousOpts_ above - using that for comparison will make it clearer.